src/domain/hanke/edit/HankeFormYhteystiedot.tsx

```diff
@@ -116,13 +116,20 @@ const ContactField: React.FC<{
   field: keyof HankeContact;
   fieldName: string;
   contactType: HankeContactTypeKey;
-}> = ({ field, fieldName, contactType }) => {
+  index: number;
+}> = ({ field, fieldName, contactType, index }) => {
   const { t } = useTranslation();
-  const { watch } = useFormContext();
-  const selectedContactType = watch(`${contactType}.0.tyyppi`);
+  const { watch, setValue } = useFormContext();
+  const selectedContactType = watch(`${contactType}.${index}.tyyppi`);
   const inputDisabled =
     field === CONTACT_FORMFIELD.TUNNUS && selectedContactType === CONTACT_TYYPPI.YKSITYISHENKILO;
 
+  useEffect(() => {
+    if (inputDisabled) {
+      setValue(`${contactType}.0.ytunnus`, null);
```

Comment:
{ shouldValidate: true } should be added as third argument to setValue so that validation is done after setting the value and possible validation errors go away.

Reply:
Fixed.